src/models/user.model.js

```diff
@@ -35,37 +32,20 @@ const userSchema = new Schema({
   active: {
     type: Boolean,
     default: false
-  },
-  role: {
-    type: String,
-    default: 'user',
-    enum: roles
   }
 }, {
   timestamps: true
 })
 
-userSchema.pre('save', async function save (next) {
-  try {
-    if (!this.isModified('password')) {
-      return next()
-    }
-
-    this.password = bcrypt.hashSync(this.password)
-
-    return next()
-  } catch (error) {
-    return next(error)
-  }
-})
+hashPass(userSchema)
 
 userSchema.post('save', async function saved (doc, next) {
   try {
     const mailOptions = {
       from: 'noreply',
       to: this.email,
       subject: 'Confirm creating account',
-      html: `<div><h1>Hello new user!</h1><p>Click <a href="${config.hostname}/api/auth/confirm?key=${this.activationKey}">link</a> to activate your new account.</p></div><div><h1>Hello developer!</h1><p>Feel free to change this template ;).</p></div>`
+      html: `<div><h1>Hello new user!</h1><p>Click <a href="${config.hostname}/api/user/confirm?key=${this.activationKey}">link</a> to activate your new account.</p></div><div><h1>Hello developer!</h1><p>Feel free to change this template ;).</p></div>`
```

Comment:
config.hostname is not available, this will not work

Reply:
I added a config param. Check in config file

Reply:
[Look here](https://github.com/kasvith/express-mongo-jwt-boilerplate/blob/b79aa1bcf3307d51744469d6919d62da3bc91441/src/config/index.js#L8)

But it seems I've forgotten to add it to .env.example somehow

Reply:
Oh I didn't know but wanted to keep this changed and renamed it to BASE_URI in next PR. Sorry for confusion.

Reply:
I added hostname because imagine we are running in something like kubernets. In that scenario when sending emails we should use the web address not the internal IP.

But I see the name is more confusing, we should rename it to something like `siteUrl` which makes more sense

Reply:
I agree, we can rename it and when this PR will be merged I'll resolve conflicts in second one.